syntax = "proto3";

package mlservice.v1;

service PredictionService {
  rpc Health(HealthRequest) returns (HealthResponse);
  rpc Predict(PredictRequest) returns (PredictResponse);
}

message HealthRequest {}

message HealthResponse {
  string status = 1;        // "ok"
  string model_version = 2;
}

message PredictRequest {
  float sepal_length = 1;
  float sepal_width = 2;
}

message PredictResponse {
  string prediction = 1;
  float confidence = 2;
  string model_version = 3;
}
